core/src/main/scala/tofu/Context.scala

```diff
@@ -74,50 +110,44 @@ trait Provide[F[_]] {
     new ProvideExtractInstance[F, Lower, Ctx, A](this, extract)
 }
 
-object Provide extends ProvideInstances[HasProvide] {
+object Provide {
   def apply[F[_]](implicit p: Provide[F]): HasProvide[F, p.Lower, p.Ctx] = p
   type Aux[F[_], G[_], C] = HasProvide[F, G, C]
+
+  final implicit def readerTContext[F[_]: Applicative, C]: HasProvide[ReaderT[F, C, *], F, C] =
+    ContextBase.readerTContext[F, C]
 }
 
-trait WithProvide[F[_], G[_], C] extends Provide[F] with Lift[G, F]    {
+trait WithProvide[F[_], G[_], C] extends Provide[F] with Lift[G, F] {
   override type Lower[A] = G[A]
   override type Ctx      = C
 }
-object WithProvide               extends ProvideInstances[WithProvide] {
+object WithProvide {
   def apply[F[_], G[_], C](implicit p: WithProvide[F, G, C]): WithProvide[F, G, C] = p
 }
 
-trait ProvideInstances[TCA[f[_], g[_], r] >: WithProvide[f, g, r]] extends RunContextInstances[TCA]
-
 trait RunContext[F[_]] extends Local[F] with Provide[F] {
   def unlift: F[F ~> Lower]                                          = ask(ctx => funK(runContext(_)(ctx)))
   def runEquivalent[A](eq: Equivalent[Ctx, A]): WithRun[F, Lower, A] =
     new RunContextEquivalentInstance[F, Lower, Ctx, A](this, eq)
 }
 
-object RunContext extends RunContextInstances[HasContextRun] {
+object RunContext {
   def apply[F[_]](implicit ctx: RunContext[F]): HasContextRun[F, ctx.Lower, ctx.Ctx] = ctx
   type Aux[F[_], G[_], C] = HasContextRun[F, G, C]
+
+  final implicit def readerTContext[F[_]: Applicative, C]: HasContextRun[ReaderT[F, C, *], F, C] =
```

Comment:
So we still can't avoid this even with Kai's approach?

Reply:
Why should we avoid this?

Reply:
Yes, unfortunately, scala reject to search parent companions for Aux-types with lifted higher-kinded type-members.
So, removing this and `Provide`'s one makes [ContextSuite](https://github.com/TinkoffCreditSystems/tofu/blob/master/core/src/test/scala/tofu/ContextSuite.scala) fail